spotfinder/spotfinder.cc

```diff
@@ -879,7 +878,7 @@ int main(int argc, char **argv) {
                         centers_of_mass.push_back(y);
                         centers_of_mass.push_back(z);
                     }
-                    if (save_to_h5) {
+                    if (save_to_h5 | output_for_index) {
```

Comment:
In the case of output_for_index=true, save_to_h5=false, we only call the 2d_components for each image in order to pipe the reflection centroids onwards, we don't need to store them for saving. So we don't need to save into the map here (and therefore also don't need to create it in L667)?